Tests/MaterialLib/TestMPLParseMaterial.cpp

```diff
@@ -0,0 +1,274 @@
+/**
+ * \file
+ * \author Norbert Grunwald
+ * \date   Sep 22, 2017
+ *
+ * \copyright
+ * Copyright (c) 2012-2019, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include <gtest/gtest.h>
+
+#include <sstream>
+
+#include "TestMPL.h"
+#include "Tests/TestTools.h"
+
+#include "MaterialLib/MPL/Medium.h"
+
+namespace MPL = MaterialPropertyLib;
+
+/// A simple structure for a string-based component used to create
+/// an XML-Structure.
+struct Component
+{
+    std::vector<std::string> property;
+    Component() : property(MPL::number_of_properties) {}
+};
+/// A simple structure for a string-based phase used to create
+/// an XML-Structure.
+struct Phase
+{
+    std::vector<Component> component;
+    std::vector<std::string> property;
+    Phase(std::size_t componentNumber)
+        : component(componentNumber), property(MPL::number_of_properties)
+    {
+    }
+};
+/// A simple structure for a string-based medium used to create
+/// an XML-Structure
+struct Medium
+{
+    std::vector<Phase> phases;
+    std::vector<std::string> property;
+    Medium(std::vector<std::size_t> topology)
+        : property(MPL::number_of_properties)
+    {
+        for (auto p : topology)
+            phases.push_back(Phase(p));
+    }
+};
+
+/// Short method creating a number of blanks used for indentation.
+std::string indent(std::size_t ind)
+{
+    return std::string(ind, ' ');
+}
+
+/// This method creates an XML-snippet of a sigle property.
+std::string makeProperty(std::size_t ind, std::size_t index,
```

Comment:
Why not `... makeProperty(std::size_t const ind, std::size_t const index, std::string const& property)`?